tests/Serializer/JsonSerializationTest.php

```diff
@@ -120,7 +120,7 @@ protected static function getContent($key)
             $outputs['post'] = '{"type":"post","title":"Post Title"}';
             $outputs['image_post'] = '{"type":"image_post","title":"Image Post Title"}';
             $outputs['image_post_without_type'] = '{"title":"Image Post Title"}';
-            $outputs['garage'] = '{"vehicles":[{"km":3,"type":"car"},{"km":1,"type":"moped"}]}';
+            $outputs['garage'] = '{"vehicles":[{"km":3,"type":"car"},{"km":1,"type":"moped"},{"km":1,"type":"JMS\\\Serializer\\\Tests\\\Fixtures\\\Discriminator\\\Other"}]}';
```

Comment:
Just thinking about the use cases. I do wonder if we should also provide some name like we have in the map - as FQN class names looks weird with other names.

Reply:
oh, it second issue with determinator - without this is hard to migrate old code to new with use determinator - need change all code in once

```
'The discriminator field name "%s" of the base-class "%s" conflicts with a regular property of the sub-class "%s".',
```

why can not use exist field in base class as determinator?

For example now in our site we have cart with many products and we use one class CartItem with field type

Reply:
hm, or all this poll have sence only with implement virtual determinator?

Reply:
Good question. May @goetas have some ideas how it should work, as he has much more experience with discriminators.